generator/python/build.gradle

```diff
@@ -49,39 +50,43 @@ task generateClient(type: GradleBuild) {
-
-        def generateTask = task("openApiGenerate_python_" + parsedApiName, type: GenerateTask.class) {
+        def namespaceBase = "${criteoPackage}.${criteoService}.${versionInNamespace}".replace('.', nameSeparator).toString(); // example: criteo\api\marketingsolutions\v2021_10
+        
+        def artifactVersion = versionCategory == 'preview' ? '0' : criteoApiVersion.replaceAll("_",".").toString()
+        artifactVersion += ".${generatorVersion}.${getFormattedDate()}".toString() // example : 2021.01.0.211110 for stable , 0.0.211110 for preview
+ 
+        def generateTask = task("openApiGenerate_" + language + "_" + parsedApiName, type: GenerateTask.class) {
                 generatorName = "python"
                 templateDir = "$resourcesDir/templates/".toString()
                 inputSpec = specPath
                 outputDir = clientOutputDirPerVersion
                 removeOperationIdPrefix = true
                 generateApiTests = false
                 generateModelTests = false
-                gitUserId='criteo'
-                gitRepoId='criteo-api-python-sdk'
+                gitUserId = 'criteo'
+                gitRepoId = "criteo-api-${criteoService}-${versionCategory}-sdk".toString()
                 configOptions = [
-                        apiVersion              : parsedJson.info.version,
-                        packageName             : "${packageBase}".toString(),
-                        invokerPackage          : 'Criteo\\SDK',
-                        packageUrl              : 'https://github.com/criteo/criteo-api-python-sdk',
-                        hideGenerationTimestamp : 'true',
-                        artifactId              : "criteo-api-${criteoService}-sdk-${technologyStack}".toString(),
-                        packageVersion          : "${packageVersion}".toString(), // v2021_10.1.211109
                         groupId                 : 'criteo',
-                        version                 : '1.0.0',
+                        apiVersion              : "${artifactVersion}".toString(),
+                        artifactId              : "criteo-api-${criteoService}-${versionCategory}-sdk".toString(),
                         description             : "Python SDK for Criteo API ${criteoService} for ${criteoApiVersion} version".toString(),
-                        sourceCompatibility     : '1.8'
+                        hideGenerationTimestamp : 'true',
+                        invokerPackage          : 'Criteo\\SDK',
+                        packageName             : "${namespaceBase}".toString(),
+                        packageUrl              : 'https://github.com/criteo/criteo-api-${language}-sdk',
+                        packageVersion          : "${artifactVersion}".toString(), // v2021_10.1.211109
+                        sourceCompatibility     : '1.8',
+                        version                 : '1.0.0',
```

Comment:
What the "version" option is for?

Reply:
From the open api tools generator source code, "version" is used as the "API version", whereas in the default mustache, we have: 
in java: 
 - in pom, "package version" = artifactVersion
 - in readme, "API version" = appVersion ; "project version"= artifactVersion
 
in python : 
- in configuration , "Version of the API" =  version;  "SDK Package Version" = packageVersion 
- in readme, "API version" = appVersion
- in setup, "VERSION" = packageVersion

in php: 
- in composer, "version" = artifactVersion
- in readme, "API version " = appVersion,  "Package version" = artifactVersion

appVersion is retrieved from the version in the swagger (it seems not to be overridable)